src/peft/tuners/lora/model.py

```diff
@@ -193,11 +193,6 @@ def _replace_module(self, parent, child_name, new_module, child):
         if hasattr(child, "base_layer"):
             child = child.base_layer
 
-        if not hasattr(new_module, "base_layer"):
```

Comment:
Why this has been removed?

Reply:
Sorry, forgot to put this into the description of the PR.

These lines are obsolete for some time now. They only apply when we unload the model (otherwise, the `if` does not match). Remember when we made the `base_layer` switch, we ensured that when unloading, we simply return the `base_layer`, no more need to create a new layer (say, a new `nn.Linear` when using `lora.Linear`) and replace the new layer's `weight` by the parent layer's `weight`. The `base_layer` already has the original `weight`. Therefore, these lines are unnecessary.

I removed them now because they were annoying with `MultiheadAttention`, because that layer has no `weight` attribute, so this line would fail.